src/main/java/in/projecteka/consentmanager/user/SessionService.java

```diff
@@ -20,13 +24,33 @@
     private final TokenService tokenService;
     private final CacheAdapter<String, String> blacklistedTokens;
     private final Logger logger = LoggerFactory.getLogger(SessionService.class);
+    private final LockedUsersRepository lockedUsersRepository;
 
     public Mono<Session> forNew(SessionRequest request) {
         if (StringUtils.isEmpty(request.getUsername()) || StringUtils.isEmpty(request.getPassword()))
             return Mono.error(ClientError.unAuthorizedRequest());
         return tokenService.tokenForUser(request.getUsername(), request.getPassword())
                 .doOnError(error -> logger.error(error.getMessage(), error))
-                .onErrorResume(error -> Mono.error(ClientError.unAuthorizedRequest()));
+                .onErrorResume(InvalidUserNameException.class, error -> Mono.error(ClientError.unAuthorizedRequest1()))
+                .onErrorResume(error -> lockedUsersRepository.getLockedUserFor(request.getUsername())
+                        .switchIfEmpty(
+                                lockedUsersRepository.insert(
+                                        new LockedUser(1, request.getUsername(), false, null))
+                                        .then(Mono.error(ClientError.unAuthorizedRequest())))
+                        .flatMap(optionalLockedUser -> {
+                            var isLocked = false;
+                            var blockedTime = "";
+                            var clientError = ClientError.unAuthorizedRequest();
+                            if (optionalLockedUser.getInvalidAttempts() >= 5) {
+                                isLocked = true;
+                                blockedTime = new Date().toString();
+                                clientError = ClientError.userBlocked();
+                            }
+                            return lockedUsersRepository
+                                    .updateUser(isLocked, blockedTime,
+                                            optionalLockedUser.getPatientId(), optionalLockedUser.getInvalidAttempts() + 1)
+                                    .then(Mono.error(clientError));
+                        }));
```

Comment:
can you move this to separate class? locking it's just by attempts and also with duration which attempts were tried

Reply:
Yea, I'll do that

Reply:
YEa, will do that